### PR TITLE
Correctly diff exposed bundles/model that do not specify expose settings

### PR DESCRIPTION
**Pull request by achilleasa**
If a bundle (or model) only specifies the exposed flag and no
per-endpoint expose settings, the diff logic will assume an implicit
"expose all endpoints to 0.0.0.0/0" when calculating diffs. This is what
actually happens in pre-2.9 controllers and allows us to produce more
accurate diff reports. This is a followup to https://github.com/juju/bundlechanges/pull/68.

For example when we diff:

```yaml
 series: bionic
applications:
  apache2:
    charm: cs:apache2-35
    num_units: 1
    expose: true
```

and the equivalent controller model of:

```yaml
 series: bionic
applications:
  apache2:
    charm: cs:apache2-35
    num_units: 1
--- # overlay.yaml
applications:
  apache2:
    exposed-endpoints:
      website:
        expose-to-cidrs:
        - 10.0.0.0/24
```

we should get (with these changes) the following changes in the diff:

```yaml
exposed-endpoints:
  "":    <------ the changes allow us to see this important diff entry
    bundle:
       expose-to-cidrs: ["0.0.0.0/0"]
    model: null  <----- now we know that the model is not exposing everything to everyone
  website:
    bundle: null
    model:
       expose-to-cidrs: ["10.0.0.0/24"]
```

### CODE REVIEW

**Review by SimonRichardson — APPROVED** [review]

**Comment by achilleasa** [comment]
$$merge$$

File: diff.go [hunk 1]
```diff
@@ -113,6 +115,26 @@ func (d *differ) diffApplication(name string) *ApplicationDiff {
 	effectiveBundleExpose := bundle.Expose || len(bundle.ExposedEndpoints) != 0
 	effectiveModelExpose := model.Exposed || len(model.ExposedEndpoints) != 0
 
+	// If any of the sides is exposed but lacks any expose endpoint details
+	// assume an implicit expose "" to 0.0.0.0/0 for comparison purposes.
```
Comment (manadart) [inline diff.go:119]: Can we still make this assumption?

Comment (achilleasa) [reply diff.go:119]: I believe this is needed this for backwards compatibility when calculating diffs with pre 2.9 bundles that only specify `expose: true`. 

Note that the migration step in the linked PR will inject 0.0.0.0/0 and ::/0 for the wildcard endpoint for any exposed app. Having this extra logic here allows us to correctly perform the diff regardless of which controller version provided the current model state to diff against. 

It's kind of an edge case but it allows us to produce consistent diffs.

Comment (achilleasa) [reply diff.go:119]: Also, there is a followup PR (https://github.com/juju/bundlechanges/pull/71) that updates this to also include the IPV6 CIDRs

**Pull request merged**